### DISCUSSION TITLE
Changelog generation is extremely slow

### DISCUSSION THREAD

**Discussion by raghunandannm** [pawamoy/git-changelog#81] [Q&A]
Hi Team, on an extremely huge repository, the git-changelog is taking too long to generate the changelog. Is there any way to speeden up the process especially working on WSL

**Comment by pawamoy** [comment]
Hello, thanks for report! Did you use git-changelog before and it was fast enough, or did you just try it out now?

In any case I have some idea to greatly speed things up :slightly_smiling_face: I'll post updates here as soon as I get some PRs in draft mode.

Ah, could you also share a link to your repository if it is public? That will help us profile perfs.

In the meantime you can try filtering commits: https://pawamoy.github.io/git-changelog/usage/#filter-commits.

**Reply by raghunandannm** [reply]
Thanks for the quick response. It seems the issue is only when working on WSL. On VM it works really quick in comparison.

**Reply by pawamoy** [reply]
I see, thanks. Unfortunately I don't have a machine with the annoying operating system installed, so I won't be able to investigate slowness in WSL. But we still have room for speedups regardless. You can also try https://github.com/orhun/git-cliff which seems similar in features to git-changelog, and is written in Rust so probably much more performant :slightly_smiling_face: I don't think it supports PEP 440 though.
